syntax = "proto3";

package pgv;

import "validate/validate.proto";

message Small {
    uint32 field1 = 1 [(validate.rules).uint32 = { gt: 0, lte: 10 }];     
    uint32 field2 = 2 [(validate.rules).uint32 = { gt: 0, lte: 10 }];     
    uint32 field3 = 3 [(validate.rules).uint32 = { gt: 0, lte: 10 }];     
    uint32 field4 = 4 [(validate.rules).uint32 = { gt: 0, lte: 10 }];     
    uint32 field5 = 5 [(validate.rules).uint32 = { gt: 0, lte: 10 }];   
      
    uint32 field6 = 6 [(validate.rules).uint32 = { gt: 0, lte: 50 }];     
    uint32 field7 = 7 [(validate.rules).uint32 = { gt: 0, lte: 50 }];     
    uint32 field8 = 8 [(validate.rules).uint32 = { gt: 0, lte: 50 }];     
    uint32 field9 = 9 [(validate.rules).uint32 = { gt: 0, lte: 50 }];     
    uint32 field10 = 10 [(validate.rules).uint32 = { gt: 0, lte: 50 }];     

    uint64 field31 = 31 [(validate.rules).uint64 = { gt: 20, lte: 2500000000 }];   
    uint64 field32 = 32 [(validate.rules).uint64 = { gt: 20, lte: 2500000000 }];   

    MessageB field41 = 41;
    MessageC field42 = 42;
    MessageD field43 = 43;
    MessageE field44 = 44;
    MessageF field45 = 45;

    bytes field51 = 51 [(validate.rules).bytes.max_len = 10];  

    sint32 field101 = 101 [(validate.rules).sint32 = {gt: -100, lte: 100}];

    message MessageB {
        uint32 field1 = 1 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field2 = 2 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field3 = 3 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        MessageE field4 = 4;
        MessageF field5 = 5;
    }

    message MessageC {
        uint32 field1 = 1 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field2 = 2 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field3 = 3 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        MessageE field4 = 4;
        MessageF field5 = 5;
    }

    message MessageD {
        uint32 field1 = 1 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field2 = 2 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field3 = 3 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        MessageE field4 = 4;
        MessageF field5 = 5;
    }

    message MessageE {
        uint32 field1 = 1 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field2 = 2 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field3 = 3 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field4 = 4 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        MessageF field5 = 5;
    }

    message MessageF {
        uint32 field1 = 1 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field2 = 2 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field3 = 3 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field4 = 4 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
        uint32 field5 = 5 [(validate.rules).uint32 = { gt: 0, lte: 1000 }]; 
    }
}
